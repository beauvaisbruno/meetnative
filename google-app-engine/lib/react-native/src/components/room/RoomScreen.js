"use strict";
var __importStar =
  (this && this.__importStar) ||
  function(mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod)
        if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
  };
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
const native_base_1 = require("native-base");
const PageTemplate_1 = __importDefault(require("../PageTemplate"));
const lang_1 = __importDefault(require("./lang"));
const reducerHelper_1 = require("../../utils/reducerHelper");
const react_redux_1 = require("react-redux");
const roomReducer_1 = require("./roomReducer");
const utilsUi_1 = require("../../utils/utilsUi");
const ChatHeader_1 = __importDefault(require("./ChatHeader"));
const react_navigation_1 = require("react-navigation");
const MessagesUi_1 = __importDefault(require("./MessagesUi"));
const react_native_firebase_1 = __importDefault(
  require("react-native-firebase")
);
const custom_1 = __importDefault(
  require("../../config/native-base-theme/variables/custom")
);
class RoomScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = Object.assign({ message: "" }, props.storyState, {
      contactProfile: this.props.navigation.getParam("contactProfile", null)
    });
  }
  componentWillUnmount() {
    this.focusListener.remove();
    this.blurListener.remove();
  }
  componentDidMount() {
    this.userId = react_native_firebase_1.default.auth().currentUser.uid;
    this.focusListener = this.props.navigation.addListener(
      "willFocus",
      payload => {
        this.setState(
          Object.assign({}, this.state, {
            contactProfile: this.props.navigation.getParam(
              "contactProfile",
              null
            )
          })
        );
        this.props.doSyncMessages();
      }
    );
    this.blurListener = this.props.navigation.addListener(
      "willBlur",
      payload => {
        this.props.doStopSyncMessages();
      }
    );
  }
  render() {
    return React.createElement(
      native_base_1.View,
      { style: { flex: 1 } },
      React.createElement(ChatHeader_1.default, {
        profile: this.state.contactProfile,
        user: this.props.userProfile
      }),
      React.createElement(MessagesUi_1.default, {
        contactProfile: this.state.contactProfile
      }),
      React.createElement(
        native_base_1.View,
        {
          style: Object.assign(
            { flexDirection: "row" },
            utilsUi_1.border(custom_1.default.borderColor),
            utilsUi_1.margin(5),
            utilsUi_1.padding(5),
            { marginTop: 0, borderRadius: 10 }
          )
        },
        React.createElement(native_base_1.Input, {
          style: Object.assign({}, utilsUi_1.noMarPad, { fontSize: 25 }),
          placeholder: lang_1.default("Message"),
          placeholderTextColor: custom_1.default.textGreyLight,
          onChangeText: text =>
            this.setState(Object.assign({}, this.state, { message: text })),
          value: this.state.message
        }),
        React.createElement(
          native_base_1.Button,
          {
            block: true,
            transparent: true,
            onPress: () => this.sendMessage(),
            testID: "sendBtn",
            style: Object.assign({}, utilsUi_1.noMarPad)
          },
          this.props.sendMessage.fetching ||
            React.createElement(native_base_1.Icon, {
              name: "send",
              type: "FontAwesome",
              style: Object.assign({}, utilsUi_1.noMarPad, {
                paddingLeft: 15,
                paddingRight: 15,
                fontSize: 40
              })
            }),
          this.props.sendMessage.fetching &&
            React.createElement(native_base_1.Icon, {
              name: "send-o",
              type: "FontAwesome",
              style: Object.assign({}, utilsUi_1.noMarPad, {
                paddingLeft: 15,
                paddingRight: 15,
                fontSize: 40
              })
            })
        )
      )
    );
  }
  sendMessage() {
    console.log("sendMessage: ", this.state.message);
    if (this.state.message === "") return;
    const msg = this.state.message;
    this.setState(Object.assign({}, this.state, { message: "" }));
    this.props.doSendMessages({
      message: msg,
      time: Date.now(),
      authorId: this.userId,
      recipientId: this.state.contactProfile.id
    });
  }
}
exports.RoomScreen = RoomScreen;
exports.mapDispatchToProps = dispatch => {
  return {
    doSendMessages: message =>
      dispatch(
        reducerHelper_1.fetchingAction(
          roomReducer_1.RoomActions.sendMessages,
          message
        )
      ),
    doSyncMessages: () =>
      dispatch(
        reducerHelper_1.fetchingAction(roomReducer_1.RoomActions.syncMessages)
      ),
    doStopSyncMessages: () =>
      dispatch(
        reducerHelper_1.fetchingAction(
          roomReducer_1.RoomActions.stopSyncMessages
        )
      )
  };
};
exports.mapStateToProps = state => ({
  sendMessage: state.room.sendMessages,
  userProfile: state.profile.profileLoad.data
});
exports.default = react_redux_1.connect(
  exports.mapStateToProps,
  exports.mapDispatchToProps
)(
  react_navigation_1.withNavigation(
    PageTemplate_1.default(RoomScreen, lang_1.default("Chat"), false, true)
  )
);
//# sourceMappingURL=RoomScreen.js.map
