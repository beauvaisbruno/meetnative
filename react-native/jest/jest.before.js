jest.mock("react-native-localize", () => ({
  findBestAvailableLanguage: () => "en"
}));
