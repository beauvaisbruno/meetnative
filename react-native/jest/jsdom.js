"use strict";
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
const jsdom = function() {
  try {
    const { JSDOM } = require("jsdom");
    const jsdom = new JSDOM("<!doctype html><html><body></body></html>");
    const { window } = jsdom;
    // @ts-ignore
    function copyProps(src, target) {
      Object.defineProperties(
        target,
        Object.assign(
          {},
          Object.getOwnPropertyDescriptors(src),
          Object.getOwnPropertyDescriptors(target)
        )
      );
    }
    // @ts-ignore
    global.window = window;
    // @ts-ignore
    global.document = window.document;
    // @ts-ignore
    global.navigator = {
      userAgent: "node.js"
    };
    copyProps(window, global);
    // @ts-ignore
    global.fetch = require("jest-fetch-mock");
    const originalConsoleError = console.error;
    console.error = message => {
      if (message.startsWith("Warning:")) {
        return;
      }
      originalConsoleError(message);
    };
  } catch (error) {
    console.log("JSDOM error: ", error);
  }
};
exports.default = jsdom;
