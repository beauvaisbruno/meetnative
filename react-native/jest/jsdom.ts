/* eslint-disable */

const jsdom = function() {
  try {
    const { JSDOM } = require("jsdom");
    const jsdom = new JSDOM("<!doctype html><html><body></body></html>");
    const { window } = jsdom;

    // @ts-ignore
    function copyProps(src: any, target: any) {
      Object.defineProperties(target, {
        ...Object.getOwnPropertyDescriptors(src),
        ...Object.getOwnPropertyDescriptors(target)
      });
    }

    // @ts-ignore
    global.window = window;
    // @ts-ignore
    global.document = window.document;
    // @ts-ignore
    global.navigator = {
      userAgent: "node.js"
    };
    copyProps(window, global);
    // @ts-ignore
    global.fetch = require("jest-fetch-mock");

    const originalConsoleError = console.error;
    console.error = (message: any) => {
      if (message.startsWith("Warning:")) {
        return;
      }
      originalConsoleError(message);
    };
  } catch (error) {
    console.log("JSDOM error: ", error);
  }
};
export default jsdom;
