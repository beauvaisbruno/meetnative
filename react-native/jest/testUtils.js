"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
// @ts-ignore
const enzyme_1 = require("enzyme");
//tree.findWhere((node => node.prop("testID") === "prop")).exists();
function expectTestIDPropExists(tree, prop) {
  let treefindWhere;
  if (tree instanceof enzyme_1.ReactWrapper)
    treefindWhere = tree.findWhere(node => node.prop("testID") === prop);
  else treefindWhere = tree.findWhere(node => node.prop("testID") === prop);
  expect(treefindWhere.exists()).toEqual(true);
}
exports.expectTestIDPropExists = expectTestIDPropExists;
function expectExists(tree, component) {
  expect(tree.find(component).exists()).toEqual(true);
}
exports.expectExists = expectExists;
const mathCopy = Object.create(global.Math);
function mockMath(rndValue = 0.5) {
  const mockMath = Object.create(global.Math);
  mockMath.random = () => 0.5;
  global.Math = mockMath;
  return rndValue;
}
exports.mockMath = mockMath;
function unMockMath() {
  global.Math = mathCopy;
}
exports.unMockMath = unMockMath;
const nowCopy = Date.now;
// const dateCopy = Object.create(global.Date);
function mockDate(date = 1546300800000 /*01/01/2019*/) {
  // const mockDate = Object.create(global.Date);
  Date.now = () => date;
  // global.Date = mockDate;
  return date;
}
exports.mockDate = mockDate;
function unMockDate() {
  // global.Date = dateCopy;
  Date.now = nowCopy;
}
exports.unMockDate = unMockDate;
