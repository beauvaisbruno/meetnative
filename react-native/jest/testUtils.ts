// @ts-ignore
import { ReactWrapper, ShallowWrapper } from "enzyme";

//tree.findWhere((node => node.prop("testID") === "prop")).exists();
export function expectTestIDPropExists(
  tree: ShallowWrapper | ReactWrapper,
  prop: string
) {
  let treefindWhere;
  if (tree instanceof ReactWrapper)
    treefindWhere = tree.findWhere(node => node.prop("testID") === prop);
  else treefindWhere = tree.findWhere(node => node.prop("testID") === prop);
  expect(treefindWhere.exists()).toEqual(true);
}
export function expectExists(
  tree: ShallowWrapper | ReactWrapper,
  component: string
) {
  expect(tree.find(component).exists()).toEqual(true);
}

const mathCopy = Object.create(global.Math);
export function mockMath(rndValue = 0.5) {
  const mockMath = Object.create(global.Math);
  mockMath.random = () => 0.5;
  global.Math = mockMath;
  return rndValue;
}

export function unMockMath() {
  global.Math = mathCopy;
}

const nowCopy = Date.now;
// const dateCopy = Object.create(global.Date);
export function mockDate(date = 1546300800000 /*01/01/2019*/) {
  // const mockDate = Object.create(global.Date);
  Date.now = () => date;
  // global.Date = mockDate;
  return date;
}

export function unMockDate() {
  // global.Date = dateCopy;
  Date.now = nowCopy;
}
