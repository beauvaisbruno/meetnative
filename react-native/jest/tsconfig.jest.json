{
  "extends": "../tsconfig",
  "compilerOptions": {
    "noImplicitAny": false
  }
}
