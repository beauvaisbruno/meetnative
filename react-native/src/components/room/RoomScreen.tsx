import * as React from "react";
import {
  Button,
  Content,
  Icon,
  Input,
  Item,
  Label,
  Text,
  View
} from "native-base";
import PageTemplate from "../PageTemplate";
import lang from "./lang";
import { IApplicationState, IProfile } from "../../rootReducers";
import { Dispatch } from "redux";
import {
  ActionResult,
  fetchingAction,
  IActionResult
} from "../../utils/reducerHelper";
import { connect } from "react-redux";
import { IMessage, RoomActions } from "./roomReducer";
import { border, font, margin, noMarPad, padding } from "../../utils/utilsUi";
import ChatHeader from "./ChatHeader";
import {
  NavigationEventSubscription,
  NavigationInjectedProps,
  withNavigation
} from "react-navigation";
import MessagesUi from "./MessagesUi";
import firebase from "react-native-firebase";
import custom from "../../config/native-base-theme/variables/custom";

export type IProps = NavigationInjectedProps & {
  doSendMessages: (message: IMessage) => void;
  doSyncMessages: () => void;
  userProfile: IProfile;
  sendMessage: IActionResult<undefined>;
  doStopSyncMessages: () => void;
};
export interface IState {
  message: string;
  contactProfile: IProfile;
}
export class RoomScreen extends React.Component<IProps, IState> {
  constructor(props: any) {
    super(props);
    this.state = {
      message: "",
      ...props.storyState,
      contactProfile: this.props.navigation.getParam("contactProfile", null)
    };
  }
  focusListener: NavigationEventSubscription | undefined;
  blurListener: NavigationEventSubscription | undefined;
  public componentWillUnmount() {
    if (this.focusListener) this.focusListener.remove();
    if (this.blurListener) this.blurListener.remove();
  }
  public componentDidMount() {
    this.focusListener = this.props.navigation.addListener(
      "willFocus",
      payload => {
        this.setState({
          ...this.state,
          contactProfile: this.props.navigation.getParam("contactProfile", null)
        });
        this.props.doSyncMessages();
      }
    );
    this.blurListener = this.props.navigation.addListener(
      "willBlur",
      payload => {
        this.props.doStopSyncMessages();
      }
    );
  }

  render() {
    return (
      <View style={{ flex: 1 }}>
        <ChatHeader
          profile={this.state.contactProfile!}
          user={this.props.userProfile}
        />

        <MessagesUi contactProfile={this.state.contactProfile!} />

        <View
          style={{
            flexDirection: "row",
            ...border(custom.borderColor),
            ...margin(5),
            ...padding(5),
            marginTop: 0,
            borderRadius: 10
          }}
        >
          <Input
            style={{ ...noMarPad, fontSize: 25 }}
            placeholder={lang("Message")}
            placeholderTextColor={custom.textGreyLight}
            onChangeText={(text: string) =>
              this.setState({ ...this.state, message: text })
            }
            value={this.state.message}
          />
          <Button
            block
            transparent
            onPress={() => this.sendMessage()}
            testID={"sendBtn"}
            style={{ ...noMarPad }}
          >
            {this.props.sendMessage.fetching || (
              <Icon
                name="send"
                type="FontAwesome"
                style={{
                  ...noMarPad,
                  paddingLeft: 15,
                  paddingRight: 15,
                  fontSize: 40
                }}
              />
            )}
            {this.props.sendMessage.fetching && (
              <Icon
                name="send-o"
                type="FontAwesome"
                style={{
                  ...noMarPad,
                  paddingLeft: 15,
                  paddingRight: 15,
                  fontSize: 40
                }}
              />
            )}
          </Button>
        </View>
      </View>
    );
  }

  private sendMessage() {
    if (this.state.message === "") return;
    const msg = this.state.message;
    this.setState({ ...this.state, message: "" });
    this.props.doSendMessages({
      message: msg,
      time: Date.now(),
      authorId: this.props.userProfile.id!,
      recipientId: this.state.contactProfile.id!
    });
  }
}
export const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    doSendMessages: (message: IMessage) =>
      dispatch(fetchingAction(RoomActions.sendMessages, message)),

    doSyncMessages: () => dispatch(fetchingAction(RoomActions.syncMessages)),
    doStopSyncMessages: () =>
      dispatch(fetchingAction(RoomActions.stopSyncMessages))
  };
};

export const mapStateToProps = (state: IApplicationState) => ({
  sendMessage: state.room.sendMessages,
  userProfile: state.profile.profileLoad.data
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withNavigation(PageTemplate(RoomScreen, lang("Chat"), false, true)));
