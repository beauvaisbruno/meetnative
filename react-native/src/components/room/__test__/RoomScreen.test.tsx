import { mockDate, unMockDate } from "../../../../jest/testUtils";

const addListener = jest.fn();
const doSendMessages = jest.fn();
const doSyncMessages = jest.fn();
const getParam = jest.fn();

jest.mock("react-native-firebase", () => {
  return {
    auth: () => ({ currentUser: { uid: "uid" } })
  };
});

import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import * as React from "react";
import { RoomScreen } from "../RoomScreen";

const initialProps = {
  userProfile: {
    id: "C3iSMgBhdmazzcOngiTBpo7USNv2",
    langsToLearn: {
      french: "intermediate",
      italian: "beginner",
      spanish: "fluent"
    },
    age: 23,
    lat: 46.144516,
    pseudo: "pseudo",
    lon: 6.081338,
    langsToTeach: {
      english: "native",
      spanish: "fluent"
    },
    geohash: 3661469831164302,
    city: "Genève",
    avatar: "cute",
    profession: "profession"
  },
  navigation: { addListener, getParam },
  sendMessage: { fetching: false },
  doSendMessages,
  doSyncMessages
};
getParam.mockReturnValue({
  langsToLearn: {
    english: "beginner",
    french: "fluent",
    german: "intermediate"
  },
  profession: "Legal Assistant",
  city: "Genève",
  geohash: 872962206,
  langsToTeach: {
    french: "fluent",
    italian: "bilingual",
    spanish: "native"
  },
  lon: 6.107057129141845,
  pseudo: "Maëlle",
  avatar: "pelican",
  lat: 46.08995392215012,
  age: 23,
  id: "IEY3I5fpxQilwKzPt0Sc"
});
it("RoomScreen", () => {
  mockDate();
  // @ts-ignore
  let tree = shallow(<RoomScreen {...initialProps} />);
  expect(toJson(tree)).toMatchSnapshot();
  expect(getParam).toBeCalledWith("contactProfile", null);
  expect(addListener.mock.calls[0][0]).toBe("willFocus");
  addListener.mock.calls[0][1]();
  expect(toJson(tree)).toMatchSnapshot();
  tree
    .find("Styled(Input)")
    .props()
    // @ts-ignore
    .onChangeText("someTextToSend");
  tree
    .find("Styled(Button)")
    .props()
    // @ts-ignore
    .onPress();
  expect(doSendMessages).toBeCalledWith({
    message: "someTextToSend",
    time: Date.now(),
    authorId: "C3iSMgBhdmazzcOngiTBpo7USNv2",
    recipientId: "IEY3I5fpxQilwKzPt0Sc"
  });
  tree = shallow(
    // @ts-ignore
    <RoomScreen {...initialProps} sendMessage={{ fetching: true }} />
  );
  expect(toJson(tree)).toMatchSnapshot();
  unMockDate();
});
