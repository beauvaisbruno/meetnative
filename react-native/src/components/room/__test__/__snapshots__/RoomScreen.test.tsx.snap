// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomScreen 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ChatHeader
    profile={
      Object {
        "age": 23,
        "avatar": "pelican",
        "city": "Genève",
        "geohash": 872962206,
        "id": "IEY3I5fpxQilwKzPt0Sc",
        "langsToLearn": Object {
          "english": "beginner",
          "french": "fluent",
          "german": "intermediate",
        },
        "langsToTeach": Object {
          "french": "fluent",
          "italian": "bilingual",
          "spanish": "native",
        },
        "lat": 46.08995392215012,
        "lon": 6.107057129141845,
        "profession": "Legal Assistant",
        "pseudo": "Maëlle",
      }
    }
    user={
      Object {
        "age": 23,
        "avatar": "cute",
        "city": "Genève",
        "geohash": 3661469831164302,
        "id": "C3iSMgBhdmazzcOngiTBpo7USNv2",
        "langsToLearn": Object {
          "french": "intermediate",
          "italian": "beginner",
          "spanish": "fluent",
        },
        "langsToTeach": Object {
          "english": "native",
          "spanish": "fluent",
        },
        "lat": 46.144516,
        "lon": 6.081338,
        "profession": "profession",
        "pseudo": "pseudo",
      }
    }
  />
  <Connect(withNavigation(MessagesUi))
    contactProfile={
      Object {
        "age": 23,
        "avatar": "pelican",
        "city": "Genève",
        "geohash": 872962206,
        "id": "IEY3I5fpxQilwKzPt0Sc",
        "langsToLearn": Object {
          "english": "beginner",
          "french": "fluent",
          "german": "intermediate",
        },
        "langsToTeach": Object {
          "french": "fluent",
          "italian": "bilingual",
          "spanish": "native",
        },
        "lat": 46.08995392215012,
        "lon": 6.107057129141845,
        "profession": "Legal Assistant",
        "pseudo": "Maëlle",
      }
    }
  />
  <Styled(ViewNB)
    style={
      Object {
        "borderBottomColor": "#a6a6a6",
        "borderBottomWidth": 1,
        "borderLeftColor": "#a6a6a6",
        "borderLeftWidth": 1,
        "borderRadius": 10,
        "borderRightColor": "#a6a6a6",
        "borderRightWidth": 1,
        "borderTopColor": "#a6a6a6",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 0,
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      }
    }
  >
    <Styled(Input)
      onChangeText={[Function]}
      placeholder="Message"
      placeholderTextColor="#919191"
      style={
        Object {
          "fontSize": 25,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
      value=""
    />
    <Styled(Button)
      block={true}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
      testID="sendBtn"
      transparent={true}
    >
      <Styled(Icon)
        name="send"
        style={
          Object {
            "fontSize": 40,
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "paddingBottom": 0,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 0,
          }
        }
        type="FontAwesome"
      />
    </Styled(Button)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`RoomScreen 2`] = `
<Styled(ViewNB)
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ChatHeader
    profile={
      Object {
        "age": 23,
        "avatar": "pelican",
        "city": "Genève",
        "geohash": 872962206,
        "id": "IEY3I5fpxQilwKzPt0Sc",
        "langsToLearn": Object {
          "english": "beginner",
          "french": "fluent",
          "german": "intermediate",
        },
        "langsToTeach": Object {
          "french": "fluent",
          "italian": "bilingual",
          "spanish": "native",
        },
        "lat": 46.08995392215012,
        "lon": 6.107057129141845,
        "profession": "Legal Assistant",
        "pseudo": "Maëlle",
      }
    }
    user={
      Object {
        "age": 23,
        "avatar": "cute",
        "city": "Genève",
        "geohash": 3661469831164302,
        "id": "C3iSMgBhdmazzcOngiTBpo7USNv2",
        "langsToLearn": Object {
          "french": "intermediate",
          "italian": "beginner",
          "spanish": "fluent",
        },
        "langsToTeach": Object {
          "english": "native",
          "spanish": "fluent",
        },
        "lat": 46.144516,
        "lon": 6.081338,
        "profession": "profession",
        "pseudo": "pseudo",
      }
    }
  />
  <Connect(withNavigation(MessagesUi))
    contactProfile={
      Object {
        "age": 23,
        "avatar": "pelican",
        "city": "Genève",
        "geohash": 872962206,
        "id": "IEY3I5fpxQilwKzPt0Sc",
        "langsToLearn": Object {
          "english": "beginner",
          "french": "fluent",
          "german": "intermediate",
        },
        "langsToTeach": Object {
          "french": "fluent",
          "italian": "bilingual",
          "spanish": "native",
        },
        "lat": 46.08995392215012,
        "lon": 6.107057129141845,
        "profession": "Legal Assistant",
        "pseudo": "Maëlle",
      }
    }
  />
  <Styled(ViewNB)
    style={
      Object {
        "borderBottomColor": "#a6a6a6",
        "borderBottomWidth": 1,
        "borderLeftColor": "#a6a6a6",
        "borderLeftWidth": 1,
        "borderRadius": 10,
        "borderRightColor": "#a6a6a6",
        "borderRightWidth": 1,
        "borderTopColor": "#a6a6a6",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 0,
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      }
    }
  >
    <Styled(Input)
      onChangeText={[Function]}
      placeholder="Message"
      placeholderTextColor="#919191"
      style={
        Object {
          "fontSize": 25,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
      value=""
    />
    <Styled(Button)
      block={true}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
      testID="sendBtn"
      transparent={true}
    >
      <Styled(Icon)
        name="send"
        style={
          Object {
            "fontSize": 40,
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "paddingBottom": 0,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 0,
          }
        }
        type="FontAwesome"
      />
    </Styled(Button)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`RoomScreen 3`] = `
<Styled(ViewNB)
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ChatHeader
    profile={
      Object {
        "age": 23,
        "avatar": "pelican",
        "city": "Genève",
        "geohash": 872962206,
        "id": "IEY3I5fpxQilwKzPt0Sc",
        "langsToLearn": Object {
          "english": "beginner",
          "french": "fluent",
          "german": "intermediate",
        },
        "langsToTeach": Object {
          "french": "fluent",
          "italian": "bilingual",
          "spanish": "native",
        },
        "lat": 46.08995392215012,
        "lon": 6.107057129141845,
        "profession": "Legal Assistant",
        "pseudo": "Maëlle",
      }
    }
    user={
      Object {
        "age": 23,
        "avatar": "cute",
        "city": "Genève",
        "geohash": 3661469831164302,
        "id": "C3iSMgBhdmazzcOngiTBpo7USNv2",
        "langsToLearn": Object {
          "french": "intermediate",
          "italian": "beginner",
          "spanish": "fluent",
        },
        "langsToTeach": Object {
          "english": "native",
          "spanish": "fluent",
        },
        "lat": 46.144516,
        "lon": 6.081338,
        "profession": "profession",
        "pseudo": "pseudo",
      }
    }
  />
  <Connect(withNavigation(MessagesUi))
    contactProfile={
      Object {
        "age": 23,
        "avatar": "pelican",
        "city": "Genève",
        "geohash": 872962206,
        "id": "IEY3I5fpxQilwKzPt0Sc",
        "langsToLearn": Object {
          "english": "beginner",
          "french": "fluent",
          "german": "intermediate",
        },
        "langsToTeach": Object {
          "french": "fluent",
          "italian": "bilingual",
          "spanish": "native",
        },
        "lat": 46.08995392215012,
        "lon": 6.107057129141845,
        "profession": "Legal Assistant",
        "pseudo": "Maëlle",
      }
    }
  />
  <Styled(ViewNB)
    style={
      Object {
        "borderBottomColor": "#a6a6a6",
        "borderBottomWidth": 1,
        "borderLeftColor": "#a6a6a6",
        "borderLeftWidth": 1,
        "borderRadius": 10,
        "borderRightColor": "#a6a6a6",
        "borderRightWidth": 1,
        "borderTopColor": "#a6a6a6",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 0,
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      }
    }
  >
    <Styled(Input)
      onChangeText={[Function]}
      placeholder="Message"
      placeholderTextColor="#919191"
      style={
        Object {
          "fontSize": 25,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
      value=""
    />
    <Styled(Button)
      block={true}
      onPress={[Function]}
      style={
        Object {
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
      testID="sendBtn"
      transparent={true}
    >
      <Component />
      <Styled(Icon)
        name="send-o"
        style={
          Object {
            "fontSize": 40,
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "paddingBottom": 0,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 0,
          }
        }
        type="FontAwesome"
      />
    </Styled(Button)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;
